append to CMAKE_MODULE_PATH instead of replace

I'm trying to use sundials in a project using CMake's FetchContent (https://cmake.org/cmake/help/latest/module/FetchContent.html). As a result I am adding the sundials directory to my CMakeLists.txt via `add_subdirectory()`.

This particular project is compiled to WASM via Emscripten, which replaces some standard CMake macros, including `CheckTypeSize`, which Sundials uses (see `SundialsIndexSize.cmake`).

In Sundials main `CMakeLists.txt`, you replace the value of CMAKE_MODULE_PATH, so in `SundialsIndexSize.cmake` the original cmake `CheckTypeSize` is called instead of the Emscripten `CheckTypeSize`, leading to the error copied below.

Would you accept a PR to append to CMAKE_MODULE_PATH rather than replace? I think this would be preferred for users wishing to include sundials via `add_subdirectory()`, unless there is a reason not to?


```
➜  build-wasm git:(main) emcmake cmake ..
configure: cmake .. -DCMAKE_TOOLCHAIN_FILE=/home/mrobins/git/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_CROSSCOMPILING_EMULATOR=/home/mrobins/git/emsdk/node/16.20.0_64bit/bin/node
cmake module path is now /home/mrobins/git/emsdk/upstream/emscripten/cmake/Modules;/home/mrobins/git/emsdk/upstream/emscripten/cmake/Modules;/home/mrobins/git/emsdk/upstream/emscripten/cmake/Modules;/home/mrobins/git/emsdk/upstream/emscripten/cmake/Modules
-- SUNDIALS_GIT_VERSION: v6.6.0
-- Looking for sys/types.h
-- Looking for sys/types.h - found
-- Looking for stdint.h
-- Looking for stdint.h - found
-- Looking for stddef.h
-- Looking for stddef.h - found
-- Check size of int64_t
-- Check size of int64_t - done
-- Check size of __int64
-- Check size of __int64 - failed
-- Check size of long
-- Check size of long - done
CMake Error at third-party/sundials-src/cmake/macros/SundialsCMakeMacros.cmake:89 (message):
  ************************************************************************

  ERROR: No integer type of size 8 was found.

                   Tried int64_t__int64longlonglong.
                   Try setting the advanced option SUNDIALS_INDEX_TYPE.

  ************************************************************************
Call Stack (most recent call first):
  third-party/sundials-src/cmake/SundialsIndexSize.cmake:45 (print_error)
  third-party/sundials-src/cmake/SundialsSetupCompilers.cmake:21 (include)
  third-party/sundials-src/CMakeLists.txt:164 (include)


-- Configuring incomplete, errors occurred!
See also "/home/mrobins/git/diffeq-backend/libs/diffeq-runtime/build-wasm/CMakeFiles/CMakeOutput.log".
See also "/home/mrobins/git/diffeq-backend/libs/diffeq-runtime/build-wasm/CMakeFiles/CMakeError.log".
emcmake: error: 'cmake .. -DCMAKE_TOOLCHAIN_FILE=/home/mrobins/git/emsdk/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake -DCMAKE_CROSSCOMPILING_EMULATOR=/home/mrobins/git/emsdk/node/16.20.0_64bit/bin/node' failed (returned 1)
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,6 +27,7 @@ project(SUNDIALS C)
 
 # Specify the location of additional CMAKE modules
 set(CMAKE_MODULE_PATH
+  ${CMAKE_MODULE_PATH}
   ${PROJECT_SOURCE_DIR}/cmake
   ${PROJECT_SOURCE_DIR}/cmake/macros
   ${PROJECT_SOURCE_DIR}/cmake/tpl

